lib/_http_server.js

```diff
@@ -475,6 +473,7 @@ function onParserExecuteCommon(server, socket, parser, state, ret, d) {
     socket.removeListener('data', state.onData);
     socket.removeListener('end', state.onEnd);
     socket.removeListener('close', state.onClose);
+    socket.removeAllListeners('drain');
```

Comment:
I'm not sure if this is ok but there are 2 listeners for the `drain` event and both seem to be useless.
The [first](https://github.com/nodejs/node/blob/5bda5faffdeefa448965159e524eacdd14490436/lib/_http_common.js#L233-L235) checks if `_httpMessage` is set and this is never true for upgrade requests.
The [second](https://github.com/nodejs/node/blob/5bda5faffdeefa448965159e524eacdd14490436/lib/_http_server.js#L376-L386) checks if the socket is `_paused` but unless I'm missing something this flag is not set for upgrade requests.